CodeLite/SocketAPI/clSocketBase.cpp

```diff
@@ -239,36 +271,28 @@ int clSocketBase::ReadMessage(wxString& message, int timeout)
     message_len = ::atoi(msglen);
 
     bytesRead = 0;
-    char* buff = new char[message_len + 1];
-    memset(buff, 0, message_len + 1);
+    std::unique_ptr<char> pBuff(new char[message_len]);
 
     // read the entire amount we need
     int bytesLeft = message_len;
     int totalRead = 0;
     while(bytesLeft > 0) {
-        rc = Read(buff + totalRead, bytesLeft, bytesRead, timeout);
+        rc = Read(pBuff.get() + totalRead, bytesLeft, bytesRead, timeout);
         if(rc != kSuccess) {
-            wxDELETEA(buff);
             return rc;
 
-        } else if(rc == 0) {

```

Comment:
Should've been a check on bytesRead == 0 presumably but instead I moved the check and throw to Read(), so the Read() used for the message_len also copes with socket being closed.